collection_mutation.hh

```diff
@@ -50,12 +54,29 @@ struct collection_mutation_view_description {
     utils::chunked_vector<std::pair<bytes_view, atomic_cell_view>> cells;
 
     // Copies the observed data, storing it in a collection_mutation_description.
-    collection_mutation_description materialize(const collection_type_impl&) const;
+    collection_mutation_description materialize(const abstract_type&) const;
+
+    // Packs the data to a serialized blob.
+    collection_mutation serialize(const abstract_type&) const;
 };
 
+// Given a linearized collection_mutation_view, returns an auxiliary struct allowing the inspection of each cell.
+// The struct is an observer of the data given by the collection_mutation_view and doesn't extend its lifetime.
+// The function needs to be given the type of stored data to reconstruct the structural information.
+collection_mutation_view_description deserialize_collection_mutation(const abstract_type&, bytes_view);
+
```

Comment:
Maybe a good way to encapsulate this is to provide a function to the deserializer to observe the data. Not suggesting to do it now, and in any case it needs to be done carefully so as not to create template bloat.

Reply:
How'd that be different from `collection_mutation_view::with_deserialized`?

Reply:
It wouldn't. I'm not sure I understand the motivation for with_deserialized().

Reply:
It's just convenient.
I still need to keep `deserialize_collection_mutation` in the header though to implement the template (I really don't want to write the implementation of `deserialize_collection_mutation` in the header)